Mathlib/Algebra/Associated.lean

```diff
@@ -240,6 +240,14 @@ theorem Irreducible.dvd_comm [Monoid α] {p q : α} (hp : Irreducible p) (hq : I
   ⟨hp.dvd_symm hq, hq.dvd_symm hp⟩
 #align irreducible.dvd_comm Irreducible.dvd_comm
 
+theorem Irreducible.of_map [Monoid R] [Monoid S] [MonoidHomClass F R S] (f : F)
+  [h : IsLocalRingHom f] {x} (hfx : Irreducible (f x)) : Irreducible x :=
+  ⟨fun hu => hfx.not_unit <| hu.map f,
+   fun p q hx =>
+    let ⟨H⟩ := h
```

Comment:
Do we not have an API for this that avoids naming the instance?

Reply:
this was done in the original proof for golf, we can just use the fully qualified name itself if that's preferrable. but this is a moved declaration that already existed